internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/storagenode/ReadComputeValidationTest.java

```diff
@@ -657,4 +657,349 @@ public void testCountGroupByValueAggregation() throws Exception {
+        departmentBuckets
+            .put("engineering", com.linkedin.venice.client.store.predicate.Predicate.equalTo("Engineering"));
+        departmentBuckets.put(
+            "any_of_eng_sales",
+            com.linkedin.venice.client.store.predicate.Predicate.anyOf("Engineering", "Sales"));
+
+        com.linkedin.venice.client.store.ComputeAggregationResponse departmentAggResponse =
+            computeStoreClient.computeAggregation()
+                .countGroupByBucket(departmentBuckets, "department")
+                .execute(keySet)
+                .get();
+
+        Map<String, Integer> departmentBucketCounts = departmentAggResponse.getBucketNameToCount("department");
+        Assert.assertNotNull(departmentBucketCounts);
+        System.out.println("Department bucket counts: " + departmentBucketCounts);
+
+        // Expected counts: engineering=2 (Engineering appears twice), any_of_eng_sales=3 (Engineering twice + Sales
+        // once)
+        Assert.assertEquals(
+            departmentBucketCounts.get("engineering"),
+            Integer.valueOf(2),
+            "Engineering count should be 2");
+        Assert.assertEquals(
+            departmentBucketCounts.get("any_of_eng_sales"),
+            Integer.valueOf(3),
+            "Any of Engineering,Sales count should be 3");
+
+        // Test complex predicate combinations
+        Map<String, com.linkedin.venice.client.store.predicate.Predicate<Integer>> complexBuckets = new HashMap<>();
+        complexBuckets.put(
+            "and_combination",
+            com.linkedin.venice.client.store.predicate.Predicate.and(
+                com.linkedin.venice.client.store.predicate.IntPredicate.greaterThan(20),
+                com.linkedin.venice.client.store.predicate.IntPredicate.lowerThan(40)));
+        complexBuckets.put(
+            "or_combination",
+            com.linkedin.venice.client.store.predicate.Predicate.or(
+                com.linkedin.venice.client.store.predicate.IntPredicate.equalTo(25),
+                com.linkedin.venice.client.store.predicate.IntPredicate.equalTo(42)));
+
+        com.linkedin.venice.client.store.ComputeAggregationResponse complexAggResponse =
+            computeStoreClient.computeAggregation().countGroupByBucket(complexBuckets, "age").execute(keySet).get();
+
+        Map<String, Integer> complexBucketCounts = complexAggResponse.getBucketNameToCount("age");
+        Assert.assertNotNull(complexBucketCounts);
+        System.out.println("Complex bucket counts: " + complexBucketCounts);
+
+        // Expected: and_combination=3 (25, 28, 35), or_combination=2 (25, 42)
+        Assert.assertEquals(
+            complexBucketCounts.get("and_combination"),
+            Integer.valueOf(3),
+            "And combination count should be 3");
+        Assert.assertEquals(
+            complexBucketCounts.get("or_combination"),
+            Integer.valueOf(2),
+            "Or combination count should be 2");
+      });
```

Comment:
We could add a test fail with predicate we cannot handle and to see if client could throw exceptions as expected.